drivers/disk/sdmmc_stm32.c

```diff
@@ -302,11 +305,13 @@ static int stm32_sdmmc_access_init(struct disk_info *disk)
 		return err;
 	}
 
+#if STM32_SDMMC_USE_RESET
```

Comment:
Nit: Using `IS_ENABLED()` to preserve readability (may have the consequence of having `.reset` always present in the structure, but if we consider we're building a SDMMC application, so it can be seen as neglectible).

This being said proposed code is aligned with DMA handling, so keeping it this way could be ok as well. This is readability vs consistency ..